borderless/explainer.md

```diff
@@ -0,0 +1,512 @@
+enabled using AWC and HTML/JavaScript/CSS (see non-goals and
+[additional-windowing-controls explainer](<[http://go/additional-windowing-controls](https://github.com/ivansandrk/additional-windowing-controls/blob/main/awc-explainer.md)>)).
+Additionally to hiding the native title bar, also the menu button is hidden.
+
+Example apps in borderless mode could look e.g. like below but the appearance
+would eventually fully depend on what the developer would implement.
+
+![Example borderless PWA](./images/borderless-app-with-AWC.png)
+
+The desire to remove the title bar will be declared within the web app manifest
+by setting the `display_override` to `borderless`. The display_override value
+will be ignored on unsupported OSs and they will follow the normal display mode
+hierarchy.
+
+```
+{
+  "display_override": [ "borderless" ]
+}
+```
+
+### 2. Enabling draggable regions
+
+When the app enters the borderless mode,
+[draggable regions](https://github.com/WICG/window-controls-overlay/blob/main/explainer.md#defining-draggable-regions-in-web-content)
+`-webkit-app-region: drag;` will be enabled.
+
+### 3. CSS display-mode media query for borderless
+
+Similarly to other display modes, borderless should be queryable with @media.
+
+Example media query with borderless:
+
+```
+@media (display-mode: borderless) {
+  .any-css-class-name {
+    margin: 5px;
+  }
+}
+```
+
+See:
+[https://developer.mozilla.org/en-US/docs/Web/CSS/@media/display-mode](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/display-mode)
+
+### 4. Leveraging Window Management permission for borderless
+
+In order for the borderless mode to activate, the app needs to have Window
+Management permission granted. The permission prompt can be triggered using
+JavaScript.
+
+For the **managed** context, permission prompts can be by-passed with a policy
+set by an admin and later on edited in the app’s settings. For the **unmanaged**
+context, the Window management API permission must be granted.
+
+#### Enabling Window Management
+
+Granting the window management API permission can be done with the following
+JavaScript:
+
+```
+const screenDetails = await window.getScreenDetails();
+```
+
+That will show the permission prompt to enable Window Management API permission
+which will also enable borderless capability in case the permission is given.
+
+Later on, the state of the permission can be queried with the following script:
+
+```
+navigator.permissions.query({name:'window-placement'})
+  .then((status) => {
+    // Do what you need with the permission state.
+    console.log(status.state)
+  });
+```
+
+#### Disabling Window Management
+
+To disable borderless mode, the user can disable the Window Management
+permission in App Settings.
+
+In case the Window management permission is not provided or no longer in place,
+the app will follow the normal display mode hierarchy based on the
+`display_override` and display values on the `manifest.json`. In case no other
```

Comment:
If no other display mode is provided, the manifest spec requires you to assume `display: browser`. There are two possibilities here:

**First possibility: You actually mean to alter the fallback chain**

I suppose you could explicitly specify "if you have `display_override: borderless` then the default `display` is `standalone` but that seems like a weird edge case. I think it makes sense to just fall back to browser, and ask developers to ask for `standalone` if they want it.

**Second possibility: Borderless always behaves like standalone if not enabled**

Alternatively, you could explain this as not "falling back to standalone mode" but rather, "borderless mode will behave like standalone if the permission is not granted". There is a key difference here: you won't use the fallback chain at all. That is, if you write `display_override: borderless, browser`, then if the UA recognizes borderless, but the permission is not granted, it will still behave like standalone. Therefore, "In case no other display mode is provided" is not true -- it will default to standalone *regardless of whether any other display modes are provided*.

I think the answer depends on the above question, of how you intend the permission to be granted (if it dynamically switches between having a title bar and borderless, then the second possibility is the best).

Reply:
It is impossible to open an Isolated Web App in browser mode so that means it is also not possible to have a borderless app that would be opened in browser mode.